simpleclient_log4j/src/main/java/io/prometheus/client/log4j/InstrumentedAppender.java

```diff
@@ -0,0 +1,93 @@
+package io.prometheus.client.log4j;
+
+import io.prometheus.client.CollectorRegistry;
+import io.prometheus.client.Counter;
+import org.apache.log4j.Appender;
+import org.apache.log4j.AppenderSkeleton;
+import org.apache.log4j.Level;
+import org.apache.log4j.spi.LoggingEvent;
+
+
+public class InstrumentedAppender extends AppenderSkeleton {
+    
+	private final CollectorRegistry registry;
+
+    private Counter all;
+    private Counter trace;
+    private Counter debug;
+    private Counter info;
+    private Counter warn;
+    private Counter error;
+    private Counter fatal;
+
+    /**
+     * Create a new instrumented appender using the default registry.
+     */
+    public InstrumentedAppender() {
+        this(CollectorRegistry.defaultRegistry);
+    }
+
+    /**
+     * Create a new instrumented appender using the given registry.
+     *
+     * @param registry the collector registry
+     */
+    public InstrumentedAppender(CollectorRegistry registry) {
+        this.registry = registry;
+        setName(Appender.class.toString());
+    }
+
+    @Override
+    public void activateOptions() {
+        this.all = createCounterForLevel("total");
+        this.trace = createCounterForLevel("trace");
+        this.debug = createCounterForLevel("debug");
+        this.info = createCounterForLevel("info");
+        this.warn = createCounterForLevel("warn");
+        this.error = createCounterForLevel("error");
+        this.fatal = createCounterForLevel("fatal");
+    }
+	
+	private Counter createCounterForLevel(String level) {
+		return Counter.build().name("log4j_appender_" + level)
```

Comment:
yeah, I'd pre-cache the labels() return value at construction time to be safe. It might be an I/O path, but many people overuse logging enough that it may be relevant.